src/twisted/names/common.py

```diff
@@ -162,14 +157,35 @@ def getHostByName(self, name, timeout=None, effort=10):
         d.addCallback(self._cbRecords, name, effort)
         return d
 
-    def _cbRecords(self, records, name, effort):
-        (ans, auth, add) = records
+    def _cbRecords(self, resp, name, effort):
+        ans, auth, add = resp.answer, resp.authority, resp.additional
         result = extractRecord(self, dns.Name(name), ans + auth + add, effort)
         if not result:
             raise error.DNSLookupError(name)
         return result
 
 
+@dataclass
+class ResolverResponse:
+    """
+    DNS message data that can be written by a resolver. This can be seen
+    as a selective view of a DNS message.
+    """
+
+    response_code: int = 0
+    """Represents the RCODE field in a DNS message header, as specified in
+    RFC1035 Section 4.1.1."""
+    answer: list[dns.RRHeader] = field(default_factory=list)
+    """Represents the answer section in a DNS message, as specified in
+    RFC1035 Section 4.1."""
+    authority: list[dns.RRHeader] = field(default_factory=list)
+    """Represents the authority section in a DNS message, as specified in
+    RFC1035 Section 4.1."""
+    additional: list[dns.RRHeader] = field(default_factory=list)
+    """Represents the additional section in a DNS message, as specified in
+    RFC1035 Section 4.1."""
+
+
```

Comment:
While I appreciate the improvements that this class represents, see previous comments about compatibilty; we really can't introduce it as such and produce a completely zero-warning compatibility earthquake for every twisted.names client.